server/sonar-web/src/main/js/apps/projectBranches/components/LongBranchesPattern.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import LongBranchesPatternForm from './LongBranchesPatternForm';
+import { getValues, SettingValue } from '../../../api/settings';
+import ChangeIcon from '../../../components/icons-components/ChangeIcon';
+import { translate } from '../../../helpers/l10n';
+
+interface Props {
+  project: string;
+}
+
+interface State {
+  formOpen: boolean;
+  loading: boolean;
+  setting?: SettingValue;
+}
+
+export const LONG_BRANCH_PATTERN = 'sonar.branch.longLivedBranches.regex';
+
+export default class LongBranchesPattern extends React.PureComponent<Props, State> {
+  mounted: boolean;
+  state: State = { formOpen: false, loading: true };
+
+  componentDidMount() {
+    this.mounted = true;
+    this.fetchSetting();
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  fetchSetting() {
+    this.setState({ loading: true });
+    getValues(LONG_BRANCH_PATTERN, this.props.project).then(
+      settings => {
+        if (this.mounted) {
+          this.setState({ loading: false, setting: settings[0] });
+        }
+      },
+      () => {
+        if (this.mounted) {
+          this.setState({ loading: false });
+        }
+      }
+    );
+  }
+
+  closeForm = () => this.setState({ formOpen: false });
+
+  handleChangeClick = (event: React.SyntheticEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    event.currentTarget.blur();
+    this.setState({ formOpen: true });
+  };
+
+  handleChange = () => {
+    if (this.mounted) {
+      this.setState({ formOpen: false });
+      this.fetchSetting();
+    }
+  };
+
+  render() {
+    const { loading, setting } = this.state;
+
+    if (!setting) {
+      return null;
+    }
+
+    return (
+      <div className="pull-right text-right">
+        <div className="little-spacer-bottom">
```

Comment:
I'd put this in `display: inline-block` to have the whole setting on one line, matching what we have to change the project visibility.

Reply:
I put it on the next line for several reasons:
* the regexp is displayed with monospaced font, which doesn't look good next to regular text
* I suspect that the regexp might be long, for example when all long living branches are specified `[branch-foo|branch-bar|branch-baz]`

Reply:
Ok then, but I still find it a bit weird to have that "setting text:" followed by two dots and being directly next to the edge of the page.

Reply:
ok, fine :)